tools/device_configuration_checker.py

```diff
@@ -296,38 +298,34 @@ def _run_tests_on_know_good_image(args, device):
         config.KNOWN_GOOD_IMAGE_FOLDER,
         device.model.lower())
 
-
-    image = None
-    if device.model.lower() == "beagleboneblack":
-        image = image_directory_path
-    elif device.model.lower() == "edison":
-        image = os.path.join(
-            image_directory_path,
-            "ostro-image-edison.ext4")
-    else:
-        image = os.path.join(
-            image_directory_path,
-            "good-image.dsk")
-
     if args.verbose:
-        print("Image file: " + str(image))
         print("Flashing " + str(device.name))
 
-    logger.info("Image file: " + str(image))
+    work_dir = device.name
+
+    # delete previous working directory, if present
+    if os.path.exists(work_dir):
+        shutil.rmtree(work_dir)
 
     try:
-        working_dir = os.getcwd()
-        os.chdir(os.path.join(image_directory_path, "iottest"))
 
-        os.chdir(image_directory_path)
+        files = get_file_list(image_directory_path)
+
+        create_work_directory(work_dir)
+        os.chdir(work_dir)
+
+        image = populate_work_directory(work_dir, image_directory_path, files)
+
+        if args.verbose:
+            print("Image file: " + image)
+        logger.info("Image file: " + str(image))
 
         device.write_image(image)
 
         tester = Tester(device)
         tester.execute()
-        results = (tester.get_results(), tester.get_results_str())
 
-        os.chdir(working_dir)
+        results = (tester.get_results(), tester.get_results_str())
 
         if not results:
```

Comment:
What if one of the 2 parts of results is null? Should that too be caught by the test or not?